cmd/argocd/commands/app.go

```diff
@@ -2463,13 +2463,44 @@ func printApplicationHistoryIds(revHistory []argoappv1.RevisionHistory) {
 // Print a history table for an application.
 func printApplicationHistoryTable(revHistory []argoappv1.RevisionHistory) {
 	w := tabwriter.NewWriter(os.Stdout, 0, 0, 2, ' ', 0)
-	_, _ = fmt.Fprintf(w, "ID\tDATE\tREVISION\n")
+	type history struct {
+		id       int64
+		date     string
+		revision string
+	}
+	varHistory := map[string][]history{}
 	for _, depInfo := range revHistory {
-		rev := depInfo.Source.TargetRevision
-		if len(depInfo.Revision) >= 7 {
-			rev = fmt.Sprintf("%s (%s)", rev, depInfo.Revision[0:7])
+		if depInfo.Source.RepoURL != "" {
+			rev := depInfo.Source.TargetRevision
+			if len(depInfo.Revision) >= 7 {
```

Comment:
Can we avoid this magic number `7` and create a constant like `MAX_ALLOWED_REVISIONS` ?

Reply:
Hi @anandf , I have removed the no 7 and added a var MAX_ALLOWED_REVISIONS